sql/database_test.go

```diff
@@ -638,3 +635,37 @@ func TestExclusive(t *testing.T) {
 		})
 	}
 }
+
+func TestConnection(t *testing.T) {
+	db := InMemoryTest(t)
+	c, err := db.Connection(context.Background())
+	require.NoError(t, err)
+	var r int
+	n, err := c.Exec("select ?", func(stmt *Statement) {
+		stmt.BindInt64(1, 42)
+	}, func(stmt *Statement) bool {
+		r = stmt.ColumnInt(0)
+		return true
+	})
+	require.NoError(t, err)
+	require.Equal(t, 1, n)
+	require.Equal(t, 42, r)
+	c.Release()
```

Comment:
If one of the assertion fails here the connection is never released. Would it make sense to defer the `Release()` call and ensure that any call besides the first is a no-op?

```suggestion
	con, err := db.Connection(context.Background())
	require.NoError(t, err)
	defer con.Release()

	var r int
	n, err := con.Exec("select ?", func(stmt *Statement) {
		stmt.BindInt64(1, 42)
	}, func(stmt *Statement) bool {
		r = stmt.ColumnInt(0)
		return true
	})
	require.NoError(t, err)
	require.Equal(t, 1, n)
	require.Equal(t, 42, r)
	require.NoError(t, con.Release())
```

Reply:
Removed this part together with `Database.Connection()`